safe-api-koa/src/index.js

```diff
@@ -44,14 +57,18 @@ export function safeGet<I: {}, O>(
   endpoint: SafeAPI.Endpoint<I, O>,
   handler: Handler<I, O>
 ): KoaMiddleware {
-  const { url, queryParams: queryParamsRep } = extractServerData(endpoint);
-  return KoaRoute.get(url, async (context, next) => {
+  const data = extractServerData(endpoint);
+  return KoaRoute.get(data.url, async (context, ...args) => {
     const input: any = {};
+    const next = args[data.lastIndex];
 
     const rawQueryParams = queryString.parse(context.request.querystring);
-    for (const key of Object.keys(queryParamsRep)) {
+    for (const key of Object.keys(data.queryParams)) {
       input[key] = rawQueryParams[key];
     }
+    for (const key of Object.keys(data.captureParamsIndex)) {
```

Comment:
Instead of having the values of `data.captureParamsIndex` be their indicies, why don't we just take advantage of the given ordering of the rows? The order in which capture params were added is reflected in the row ordering of the object. That way you also don't need to keep track of `lastIndex`, a somewhat redundant piece of data.

In the future, I can imaging the value type of this object to be a `TypeRep`, once we support non-string capture params. For now though, we're probably better off keeping this an array of keys, called `data.captureParams`, and then the order of the params is baked into the data structure. What do you think?

Reply:
Yeah, that sounds much better than what I'm doing here, thanks!